models.py

```diff
@@ -280,19 +280,38 @@ def __get_distance_weights(d, w1, w0, sigma):
     return weights
 
 
-def __get_size_weights(sizes):
-    C = Variable(torch.sqrt(torch.Tensor([sizes.size()[1] * sizes.size()[2]])) / 2, requires_grad=False)
-    if torch.cuda.is_available():
-        C = C.cuda()
+def __get_size_weights(sizes, C):
     size_weights = C / sizes
     size_weights[sizes == 1] = 1
     return size_weights
 
 
-def get_loss_params(w0, sigma):
+def get_loss_params(w0, sigma, imsize):
```

Comment:
@apyskir It seems inconsistent to build tensors here but w1 tensor is build somewhere else. Either one or the other. I think that one should simply pass constants `w0` ,`sigma`,`C`  to the `multiclass_weighted_segmentation_loss` and create all the tensors in the same function. You could have a function create_weight_variables or something if you like to (which I think is not a bad idea) but just do all that in one place.

Reply:
@jakubczakon This solution is temporary (in principle, in practice maybe it is not), because w1 should be part of a target. So building w1 is really temporary. I can move it to `get_loss_params`, but when building w1 is fixed, it will need to be changed again.